server/src/main/java/org/elasticsearch/transport/RemoteConnectionStrategy.java

```diff
@@ -126,6 +126,7 @@ public Writeable.Reader<RemoteConnectionInfo.ModeInfo> getReader() {
     protected final TransportService transportService;
     protected final RemoteConnectionManager connectionManager;
     protected final String clusterAlias;
+    protected final String authorization;
```

Comment:
we probably shouldn't have a final variable representing a dynamic setting.

As mentioned below, I don't think we the ConnectionProfile needs this information, which means that this is the variable that holds the value of the setting and it should be dynamically updatable. I believe a `addAffixUpdateConsumer`  is the method to use to ensure this gets updated properly

Reply:
I plan to remove it from profile and strategy when I know and understand which class the authorization value belongs in.

https://github.com/elastic/elasticsearch/pull/90312#discussion_r983138025

Reply:
I am saying that the variable likely belongs here in this class since it likely does not belong to the ConnectionProfile (if does not make sense here we can move it later). We should avoid representing dynamic settings with final variables and ensure that any view (variable) of the setting is updated when the setting is updated. So, even if we kept it in the profile and strategy (which i don't think we should) this variable should still support getting updated when the setting is updated.

Reply:
Thanks for the clarification. I can leave it in ConnectionProfile, and remove it from RemoteConnectionStrategy. My confusion is you previously said ApiKey was not related to connection classes, so I was waiting for clarification about some other class you had in mind.

https://github.com/elastic/elasticsearch/pull/90312#discussion_r982692117
> RemoteConnectionStrategy and ConnectionProfile are more concerned with how we make that connection and the ApiKey isn't too related to the connection itself.

Reply:
> I am saying that the variable likely belongs here in this class 

Sorry, I mispoke when I said "here in this class" (I was too quick when looking at this) I meant to re-enforce my prior [comment](https://github.com/elastic/elasticsearch/pull/90312#discussion_r982692117):

 > I think the setting should belong to RemoteClusterService or possibly RemoteClusterAwareClient . Both classes are more aligned with using an established connection for requests. RemoteConnectionStrategy and ConnectionProfile are more concerned with how we make that connection and the ApiKey isn't too related to the connection itself. 

That said, if it remains "here in this class" until we start using it that is fine too, we can always move later. The reason I object to placing it in `ConnectionProfile` is that it is not part of the connection and updates to authorization should not result in rebuilding the connection and thus we should not reuse that updater and as such this setting should have it's own way to update it's variable.

Reply:
Removed authorization from both the profile and strategy classes.